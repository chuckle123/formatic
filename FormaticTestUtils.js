import React from 'react';
import { mount } from 'enzyme';

import Formatic from './lib/formatic';

export const renderFieldsToHtml = fields => {
  const config = Formatic.createConfig();
  const value = config.createRootValue({
    fields,
  });

  const component = mount(
    <Formatic
      config={config}
      value={value}
      fields={fields}
      onChange={() => {}}
    />
  );
  const html = component.html();
  component.unmount();
  return html;
};
