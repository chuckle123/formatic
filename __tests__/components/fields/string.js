/*global describe, it, expect*/

import { renderFieldsToHtml } from '../../../FormaticTestUtils';
import stringExample from '../../../demo/examples/string';

describe('string field', () => {
  it('should render correctly', () => {
    expect(renderFieldsToHtml(stringExample.fields)).toMatchSnapshot();
  });
});
