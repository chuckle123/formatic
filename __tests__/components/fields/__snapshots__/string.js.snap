// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`string field should render correctly 1`] = `

<div class="formatic">
  <div class="optional">
    <div class>
      <span class="required-label not-required-text">
      </span>
    </div>
    <span>
    </span>
    <fieldset class>
      <div class="optional">
        <div class>
          <label>
            String
          </label>
          <span class="required-label not-required-text">
          </span>
        </div>
        <span>
        </span>
        <textarea class>
        </textarea>
      </div>
      <div class="validation-error-required required">
        <div class>
          <label>
            Note
          </label>
          <span class="required-label required-text">
          </span>
        </div>
        <span>
        </span>
        <textarea class>
        </textarea>
      </div>
      <div class="optional">
        <div class>
          <label>
            Groceries
          </label>
          <span class="required-label not-required-text">
          </span>
        </div>
        <span>
        </span>
        <div role="presentation"
             class="pretty-text-wrapper"
        >
          <div class="pretty-text-click-wrapper"
               tabindex="0"
               role="textbox"
          >
            <div role="presentation"
                 class="pretty-text-box placeholder"
                 tabindex="0"
            >
              <div class="internal-text-wrapper">
                <span>
                </span>
              </div>
            </div>
          </div>
          <a href="JavaScript:"
             class
          >
            Insert...
          </a>
        </div>
      </div>
    </fieldset>
  </div>
</div>

`;
